src/libraries/System.ComponentModel.TypeConverter/tests/Design/DesigntimeLicenseContextSerializerTests.cs

```
@@ -0,0 +1,117 @@
+                    var context = new DesigntimeLicenseContext();
+                    context.SetSavedLicenseKey(typeof(int), key);
+                    var assembly = typeof(DesigntimeLicenseContextSerializer).Assembly;
+                    Type runtimeLicenseContextType = assembly.GetType("System.ComponentModel.Design.RuntimeLicenseContext");
+                    Assert.NotNull(runtimeLicenseContextType);
+                    object runtimeLicenseContext = Activator.CreateInstance(runtimeLicenseContextType);
+                    Assert.NotNull(runtimeLicenseContext);
+
+                    Type designtimeLicenseContextSerializer = assembly.GetType("System.ComponentModel.Design.DesigntimeLicenseContextSerializer");
+                    Assert.NotNull(designtimeLicenseContextSerializer);
+
+                    using (MemoryStream stream = new MemoryStream())
+                    {
+                        long position = stream.Position;
+                        System.ComponentModel.Design.DesigntimeLicenseContextSerializer.Serialize(stream, key, context);
+                        AppContext.SetSwitch(enableBinaryFormatter, false);
+                        Reflection.MethodInfo deserializeMethod = designtimeLicenseContextSerializer.GetMethod("Deserialize", Reflection.BindingFlags.NonPublic | Reflection.BindingFlags.Static);
+                        stream.Seek(position, SeekOrigin.Begin);
+                        try
+                        {
+                            deserializeMethod.Invoke(null, new object[] { stream, key, runtimeLicenseContext });
+                        }
+                        catch (System.Reflection.TargetInvocationException exception)
+                        {
+                            var baseException = exception.GetBaseException();
+                            Assert.True(baseException is System.NotSupportedException);
+                        }
+                        catch (System.Exception catchAll)
```

This whole catch can be removed. A basic rule is, never catch an exception and then `throw caughtException;`. That kills the stack trace on the exception, losing valuable information. Always just `throw;` if you want to re-throw an exception. But here all you are doing is rethrowing, so just let the exception bubble up.